test/dynamo/test_subclasses.py

```diff
@@ -29,6 +33,177 @@
+            for o_a, o_b in zip(out_a_flat, out_b_flat)
+        ]
+        out = pytree.tree_unflatten(out_flat, spec)
+        return return_and_correct_aliasing(func, args, kwargs, out)
+
+
+class WrapperTensor(torch.Tensor):
+    @staticmethod
+    def __new__(cls, a, outer_size=None, outer_stride=None):
+        if outer_size is None:
+            outer_size = a.size()
+        if outer_stride is None:
+            outer_stride = a.stride()
+
+        shape = outer_size
+        stride = outer_stride
+        kwargs = {}
+        kwargs["strides"] = stride
+        kwargs["device"] = a.device
+        kwargs["layout"] = a.layout
+        kwargs["requires_grad"] = a.requires_grad
+        kwargs["dtype"] = a.dtype
+        out = torch.Tensor._make_wrapper_subclass(cls, shape, **kwargs)
+        return out
+
+    def __init__(self, a, outer_size=None, outer_stride=None):
+        self.a = a
+
+    def __repr__(self):
+        a_repr = repr(self.a)
+        return f"WrapperTensor({a_repr})"
+
+    def __tensor_flatten__(self):
+        return ["a"], None
+
+    @staticmethod
+    def __tensor_unflatten__(inner_tensors, meta, outer_size, outer_stride):
+        assert meta is None
+        a = inner_tensors["a"]
+        return WrapperTensor(a, outer_size, outer_stride)
+
+    @classmethod
+    def __torch_dispatch__(cls, func, types, args, kwargs):
+        if kwargs is None:
+            kwargs = {}
+        args_a = pytree.tree_map_only(WrapperTensor, lambda x: x.a, args)
+        kwargs_a = pytree.tree_map_only(WrapperTensor, lambda x: x.a, kwargs)
+
+        out_a = func(*args_a, **kwargs_a)
+
+        if (
+            out_a.numel() == 1 and out_a.dtype == torch.bool
+        ):  # short-circuit to assertEqual to work with WrapperTensor
+            return out_a
+
+        out = WrapperTensor(out_a)
+        return return_and_correct_aliasing(func, args, kwargs, out)
+
+
+class DoubleSizeTensor(torch.Tensor):
+    @staticmethod
+    def __new__(cls, inner):
+        # Double the outer-most dimension
+        if inner.numel() == 1:
+            outer_shape = ()
+        else:
+            outer_shape = (inner.shape[0] * 2,) + inner.shape[1:]
+
+        shape = outer_shape
+        kwargs = {}
+        kwargs["strides"] = inner.stride()
+        kwargs["device"] = inner.device
+        kwargs["layout"] = inner.layout
+        kwargs["requires_grad"] = inner.requires_grad
+        kwargs["dtype"] = inner.dtype
+        return torch.Tensor._make_wrapper_subclass(cls, shape, **kwargs)
+
+    def __init__(self, inner):
+        self.inner_elem = inner
+
+    def __tensor_flatten__(self):
+        return ["inner_elem"], None
+
+    @staticmethod
+    def __tensor_unflatten__(inner_tensors, meta, outer_size, outer_stride):
+        return DoubleSizeTensor(inner_tensors["inner_elem"])
+
+    def __repr__(self):
+        return f"DoubleSizeMayberAddGeThreeTensor({repr(self.inner_elem)})"
```

Comment:
nit: repr doesn't match the name (is the `AddGeThreeTensor` supposed to be part of the class name?)

Reply:
If we're going to add a bunch of one-off subclasses, what do you think of defining them all in a separate file, that way this testing file can contain mostly testing code. e.g. `torch/testing/_internal/subclass_utils.py